src/spaces/utilities.jl

```diff
@@ -255,7 +255,9 @@ function randomwalk!(
         ))
     end
     offsets = offsets_at_radius(model, r)
-    walk!(agent, rand(offsets), model)
+    # if ifempty is not set to false, `walk!` will error for GridSpaceSingle
+    ifempty = typeof(model.space) <: GridSpaceSingle ? false : true
```

Comment:
Of course this must be changed otherwise `GridSpaceSingle` will not work as expected, but currently the `walk!` function does not seem to work for these spaces? It's totally unclear to me how this is supposed to work (I hardly ever use grid models)
https://github.com/JuliaDynamics/Agents.jl/blob/ac265f9cd094684c1768e4170da70ce1cc220771/src/spaces/utilities.jl#L209
it tells me that `ids_in_position` is not defined for GridSpaceSingle; a different function should be used instead, `id_in_position` ?

Reply:
Solved this by dispatching over GridSpaceSingle